app/controllers/bookings_controller.rb

```diff
@@ -1,42 +1,69 @@
 class BookingsController < ApplicationController
   before_action :set_goat, only: [:new, :create]
+  before_action :set_booking, only: [:approve, :reject]
 
   def new
-    @booking = @goat.build_booking
+    if @goat.booking
+      redirect_to root_path, alert: 'This goat is already booked.'
+    else
+      @booking = @goat.build_booking
+    end
   end
 
   def create
     if user_signed_in?
-      # The user is logged in
-      @goat = Goat.find(params[:goat_id])
-      @booking = @goat.build_booking(booking_params)
-      @booking.status = "pending"
-      @booking.rent_start = params[:booking]["rent_start"]
-      @booking.rent_end = params[:booking]["rent_end"]
-      @booking.user = current_user
-      if @booking.rent_start && @booking.rent_end
-        days = (@booking.rent_end - @booking.rent_start).to_i
-        @booking.total_price = days * @goat.price_per_day
-      end
-
-      if @booking.save!
-        redirect_to root_path, notice: 'Booking was successfully created.'
+      if @goat.booking
+        redirect_to root_path, alert: 'This goat is already booked.'
       else
-        render :new
+        @booking = @goat.build_booking(booking_params)
+        @booking.status = "pending"
+        @booking.user = current_user
+
+        # Calculate total price based on rent_start and rent_end
```

Comment:
We have this login in the Booking model: https://github.com/RBERROU/Goatmate/blob/0a8955e3577c9ff45e1cecebc45d0629da75cc94/app/models/booking.rb#L12 
Do we need it here as well?